ml/learn.py

```diff
@@ -34,18 +34,11 @@
     import cPickle as pickle
```

Comment:
Since merlin doesn't support python2, should we drop python2 support for merlin-spellbook as well? If so, I think this is a py2 pattern

Reply:
does merlin support running python2 in the steps still (in place of bash), or is that python3 only now too?

Reply:
Good question, python2 shell does work, so I guess we should support it

Reply:
so, going the other way, do we want to support that either?  most of the 3rd party libraries have dropped python2.  looks like scikit-learn 0.20 is the last one that works in python 2, and that was released in 2018.

Reply:
My understanding is the merlin `shell` key is pretty open-ended. It will run a step in any shell you point at, so py2 isn't actually hardcoded in or explicitly supported

Reply:
This can be changed to:
try:
  import cPickle as pickle
except ImportError:
  import pickle

But the more fundamental question is will this module work if a user forces the py2 downgrade to a py2 sklearn module?

Reply:
the try ... except is already there to catch py2 issues

the functionality that this relies on is the sklearn "all_estimators" module, which has moved around some.
the try .. catch logic in surrogates.py catches older versions at least back to 0.15